yggdrasilffi/src/lib.rs

```diff
@@ -115,6 +112,25 @@ fn result_to_json_ptr<T: Serialize>(result: Result<Option<T>, FFIError>) -> *mut
     CString::new(json_string).unwrap().into_raw()
 }
 
+unsafe fn get_engine(engine_ptr: *mut c_void) -> Result<ManagedEngine, FFIError> {
+    if engine_ptr.is_null() {
+        return Err(FFIError::NullError);
+    }
+    let arc_instance = Arc::from_raw(engine_ptr as *const RawPointerDataType);
+
+    let cloned_arc = arc_instance.clone();
+    forget(arc_instance);
+
+    Ok(cloned_arc)
+}
+
+fn recover_lock<T>(lock: &Mutex<T>) -> MutexGuard<T> {
+    match lock.lock() {
+        Ok(guard) => guard,
+        Err(poisoned) => poisoned.into_inner(),
```

Comment:
A Mutex becomes poisoned when a thread panics while holding a lock to it. From Yggdrasil's perspective, either this means

- Everything is on fire and nothing will work, in which case re-accessing the poisoned Mutex won't make it worse
- Something panicked while accessing something that is safe to recover from and we can ignore the error

So that's what we do in both cases - acknowledge the poisoned state and then ignore it, by recovering with `into_inner()`